misc/dbt-materialize/dbt/include/materialize/macros/deploy/deploy_init.sql

```diff
@@ -181,7 +183,140 @@
             {% endset %}
             {{ run_query(create_cluster) }}
             {{ set_cluster_ci_tag() }}
+            {{ internal_copy_cluster_grants(schema, deploy_schema) }}
         {% endif %}
     {% endif %}
 {% endfor %}
 {% endmacro %}
+
+{% macro internal_copy_schema_default_privs(from, to) %}
+{% set revoke_initial_defaults %}
+SELECT
+  'ALTER DEFAULT PRIVILEGES ' ||
+  CASE
+    WHEN object_owner = 'PUBLIC' THEN 'FOR ALL ROLES '
+    ELSE 'FOR ROLE ' || object_owner
+  END ||
+  'IN SCHEMA quote_ident({{ to }}) ' ||
+  'REVOKE '    || privilege_type || ' ' ||
+  'ON ' || object_type || 's ' ||
+  'TO ' || grantee || ';'
+FROM mz_internal.mz_show_default_privileges
+WHERE database = current_database() AND schema = lower(trim('{{ from }}'))
+    AND object_owner <> 'none' AND grantee <> 'none'
+{% endset %}
+
+{% set revoke_defaults = run_query(revoke_initial_defaults) %}
+{% if execute %}
+    {% for alter in revoke_defaults.rows %}
+        {{ run_query(alter) }}
+    {% endfor %}
+{% endif %}
+
+{% set find_default_privs %}
+SELECT
+  'ALTER DEFAULT PRIVILEGES ' ||
+  CASE
+    WHEN object_owner = 'PUBLIC' THEN 'FOR ALL ROLES '
+    ELSE 'FOR ROLE ' || object_owner
+  END ||
+  'IN SCHEMA quote_ident({{ to }}) ' ||
+  'GRANT '    || privilege_type || ' ' ||
+  'ON ' || object_type || 's ' ||
+  'TO ' || grantee || ';'
+FROM mz_internal.mz_show_default_privileges
+WHERE database = current_database() AND schema = lower(trim('{{ from }}'))
+    AND object_owner <> 'none' AND grantee <> 'none'
+{% endset %}
+
+{% set alter_defaults = run_query(find_default_privs) %}
+{% if execute %}
+    {% for alter in alter_defaults.rows %}
+        {{ run_query(alter) }}
+    {% endfor %}
+{% endif %}
+{% endmacro %}
+
+{% macro internal_copy_schema_grants(from, to) %}
+{% set find_grants %}
+WITH schema_privledges_from AS (
```

Comment:
```suggestion
WITH schema_privilege_from AS (
```

(and in many other places below)